server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java

```diff
@@ -66,325 +62,464 @@
+    for (TsFileResource mergeTsFile : mergeTsFiles) {
+      deleteLevelFile(mergeTsFile);
+      logger.info(
+          "{} [Compaction] delete TsFile {}", storageGroupName, mergeTsFile.getTsFilePath());
     }
   }
 
-  @SuppressWarnings("squid:S1141")
-  private void updateMergeModification(TsFileResource seqFile) {
-    try {
-      // remove old modifications and write modifications generated during merge
-      seqFile.removeModFile();
-      if (mergingModification != null) {
-        for (Modification modification : mergingModification.getModifications()) {
-          // we have to set modification offset to MAX_VALUE, as the offset of source chunk may
-          // change after compaction
-          modification.setFileOffset(Long.MAX_VALUE);
-          seqFile.getModFile().write(modification);
+  public void deleteLevelFilesInList(
+      long timePartitionId, Collection<TsFileResource> mergeTsFiles, int level, boolean sequence) {
+    logger.debug("{} [compaction] merge starts to delete file list", storageGroupName);
+    if (sequence) {
+      if (sequenceTsFileResources.containsKey(timePartitionId)) {
+        if (sequenceTsFileResources.get(timePartitionId).size() > level) {
+          sequenceTsFileResources.get(timePartitionId).get(level).removeAll(mergeTsFiles);
         }
-        try {
-          seqFile.getModFile().close();
-        } catch (IOException e) {
-          logger.error(
-              "Cannot close the ModificationFile {}", seqFile.getModFile().getFilePath(), e);
+      }
+    } else {
+      if (unSequenceTsFileResources.containsKey(timePartitionId)) {
+        if (unSequenceTsFileResources.get(timePartitionId).size() > level) {
+          unSequenceTsFileResources.get(timePartitionId).get(level).removeAll(mergeTsFiles);
         }
       }
-    } catch (IOException e) {
-      logger.error(
-          "{} cannot clean the ModificationFile of {} after merge",
-          storageGroupName,
-          seqFile.getTsFile(),
-          e);
     }
   }
 
-  private void removeMergingModification() {
+  private void deleteLevelFile(TsFileResource seqFile) {
+    seqFile.writeLock();
     try {
-      if (mergingModification != null) {
-        mergingModification.remove();
-        mergingModification = null;
-      }
+      ChunkCache.getInstance().clear();
+      TimeSeriesMetadataCache.getInstance().clear();
+      FileReaderManager.getInstance().closeFileAndRemoveReader(seqFile.getTsFilePath());
+      seqFile.setDeleted(true);
+      seqFile.delete();
     } catch (IOException e) {
-      logger.error("{} cannot remove merging modification ", storageGroupName, e);
+      logger.error(e.getMessage(), e);
+    } finally {
+      seqFile.writeUnlock();
     }
   }
 
-  public void mergeEndAction(
-      List<TsFileResource> seqFiles, List<TsFileResource> unseqFiles, File mergeLog) {
-    logger.info("{} a merge task is ending...", storageGroupName);
+  public static int getMergeLevel(File file) {
```

Comment:
The current IoTDB data file format is as follows：timestamp-version-level-unseq merge cnt, is this the required merge level?